File: padroesEstruturais/padrao-de-projeto-bridge/php/tests/unit/HumanoTest.php
<?php

namespace tests;

use Bridge\Php\classes\Guerreiro;
use Bridge\Php\racas\Humano;
use PHPUnit\Framework\TestCase;

class HumanoTest extends TestCase
{
    public function testDeveRetornarPoderHumanoGuerreiro(): void
    {
        $humano = new Humano(new Guerreiro(1000));
        $this->assertEquals(1000000, $humano->calculaPoder(), 0.01);
    }
}

File: padroesEstruturais/padrao-de-projeto-bridge/java/src/test/java/HumanoTest.java
import classes.Guerreiro;
import classes.Mago;
import classes.Tank;
import org.junit.Test;
import racas.Humano;

import static org.junit.Assert.assertEquals;

public class HumanoTest {

    @Test
    public void deveRetornarPoderHumanoGuerreiro() {
        Humano humano = new Humano(new Guerreiro(10));
        assertEquals(10000, humano.calculaPoder(), 0.01);
    }

    @Test
    public void deveRetornarPoderHumanoMago() {
        Humano humano = new Humano(new Mago(100));
        assertEquals(100000, humano.calculaPoder(), 0.01);
    }

    @Test
    public void deveRetornarPoderHumanoTank() {
        Humano humano = new Humano(new Tank(1));
        assertEquals(1000, humano.calculaPoder(), 0.01);
    }
}
